neuraxle/metaopt/random.py

```diff
@@ -40,11 +40,15 @@ def __init__(self, scoring_function=r2_score, joiner=NumpyConcatenateOuterBatch(
         self.scoring_function = scoring_function
         self.joiner = joiner
 
+    def set_step(self, step):
+        MetaStepMixin.__init__(self, step)
+        return self
+
     def fit(self, data_inputs, expected_outputs=None) -> 'BaseCrossValidation':
         train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs = self.split(
             data_inputs, expected_outputs)
 
-        step = StepClonerForEachDataInput(self.step)
+        step = StepClonerForEachDataInput(self.wrapped)
```

Comment:
this is good.

Reply:
This will be tested when running the examples' tests. I think we don't have unit tests for that yet, because it would have crashed before.